### ISSUE TITLE
Repository looks dead even though it isn't quite yet

### ISSUE THREAD

**Issue by wilm0x42** [wilm0x42/wii-gc-adapter-inject#9]
stay tuned

**Comment by wilm0x42** [comment]
btw I'm on a few Project M discord servers, so feel free to ping me there if you have any questions ᕕ(ᐛ)ᕗ

**Comment by InternalLoss** [comment]
Glad to hear it's not dead, maybe add the fact you're available on certain servers into the README c:

**Comment by JacobM2** [comment]
Good to hear it is not dead! Implementation might get easier with the new loader on the horizon that does not depend on cIOS. https://bitbuilt.net/forums/index.php?threads/bbloader-v1-1-beta.3564/. The loader may get riivolution support in near future.

> Glad to hear it's not dead, maybe add the fact you're available on certain servers into the README c:


One of the servers @wilm0x42 is talking about is Project+ (https://projectplusgame.com/discord)

**Comment by geogolem** [comment]
Super excited that the repo isn't dead! :)

**Comment by wilm0x42** [comment]
![2020-11-08-013430](https://user-images.githubusercontent.com/26749441/98458822-b315e400-2162-11eb-8e06-fe614d63bff8.jpg)
One does not simply debug a wii

**Comment by InternalLoss** [comment]
is that chip your response to a lack of Gecko because if it is please make that code OSS

**Comment by wilm0x42** [comment]
Yup! Things are currently in a state where I don't believe it'll be really possible to find what's causing stuff to not work without better debugging than on-screen pixel printing.

I myself don't have the experience necesary to design a PCB for this without probably screwing it up somehow, but I found this SD Gecko board design that's close enough https://oshpark.com/shared_projects/S53fJyw8

So the current setup connects the EXI port to a SC16IS750 SPI-to-UART adapter chip, which then goes to a garden-variety UART-to-USB adapter.

It _does_ seem to work, but I've encountered a problem; the EXI hardware can communicate from 1MHz to as fast as 32MHz, and USB Gecko stuff always runs at 32MHz. This adapter can only go up to 4MHz, and unless I get an Arduino-adjacent device that runs at like 256MHz, I'd have to make it happen in an FPGA or CPLD, as did the original USB Gecko.

As such, my latest efforts have been to mess with existing wii software to try coaxing it into running EXI at a lower speed.

**Comment by Olmectron** [comment]
Hi there. Are there any updates to this? Thanks for your work.

**Comment by JacobM2** [comment]
Hey @wilm0x42  any chance you can publish any of your in progress source? It's been a bit over 2 years since the last update and your extensive research on this topic might allow someone else the springboard to a solution. Hope all is well!

**Comment by wilm0x42** [comment]
Hello! Long time no see.
Now that you mention it, I *do* have three commits I never pushed from the last time I worked on this.
Those changes modify the USB code to where it (theoretically) works on hacked versions of IOS with USB loading.

However, this left things in a strange state where it would run flawlessly in dolphin but crashes on console. (Hence my unsuccessful efforts to create a DIY USB Gecko so I could do hardware debugging)

But, as you suggest, I'll go ahead and push them now because why not.
Additionally, I actually (finally, after like 4 years of searching) was able to purchase a USB Gecko just yesterday, from @webhdx !!

Once it arrives, perhaps I'll be able to finally finish this project lol

**Comment by JacobM2** [comment]
Sweet! What a pleasant surprise! Those USB Geckos are crazy hard to find. Hope it helps with whatever else you are up to these days! Thanks again!

**Comment by Alexroseajr** [comment]
@wilm0x42 I know it's years and years and years and your interests have probably moved on, but did your USB Gecko ever help you make progress here? It sounds like it was extremely close to being finished, so tragic to see such great functionality a couple of commits away from being perfect. I've even tried the WiiU VC inject method and it hasn't worked for me, this would be incredible if it were possible
